pkg/apis/apps/validation/validation.go

```diff
@@ -371,6 +369,25 @@ func ValidateRollingUpdateDaemonSet(rollingUpdate *apps.RollingUpdateDaemonSet,
 	return allErrs
 }
 
+// validateRollingUpdateStatefulSet validates a given RollingUpdateStatefulSet.
+func validateRollingUpdateStatefulSet(rollingUpdate *apps.RollingUpdateStatefulSetStrategy, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs,
+		apivalidation.ValidateNonnegativeField(
+			int64(rollingUpdate.Partition),
+			fldPath.Child("partition"))...)
+	if rollingUpdate.MaxUnavailable != nil {
```

Comment:
done

Reply:
I disagree. The strategy would drop the field if it's disallowed. If it doesn't (because another apiserver allowed it) then we should still validate the field.

Reply:
But aren't we in the api validation stage? Out of the 2 apiservers, if one disabled and other allowed it, shouldn't other apiserver's validation logic catch it and to be fair, only apiserver can be in charge of things at a time?

Reply:
apiservers are stateless and all of them active at the same time but not sure if this is a supported configuration where different apiservers are running with different configuration/feature flags ?

Reply:
I see similar check was added for MinReadySeconds https://github.com/kubernetes/kubernetes/pull/82162/files#diff-f4270171dae1b0ad1792c3f5c5ac451e823af053462f464dd38d32fec4846b08R132

Reply:
During an upgrade, there is a time where apiservers can have different code and/or feature flags.

> Out of the 2 apiservers, if one disabled and other allowed it, shouldn't other apiserver's validation logic catch it

That is reasonable. On the other hand, we are starting to get into a pattern of avoiding to check feature gates during validation. Most validation logic doesn't require looking into the feature gate, except for some particular changes. In this case, we check the gates in the strategy and pass some validation options.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#new-enum-value-in-existing-field

Your change falls into this pattern instead: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#new-field-in-existing-api-version

Reply:
So what is the conclusion @alculquicondor @ravisantoshgudimetla should i remove the feature gate check from here ?

Reply:
Based on the guidance, yes

Reply:
done